libvmi/libvmi_extra.h

```diff
@@ -192,34 +192,20 @@ status_t vmi_get_struct_field_type_name_from_json(
     const char **member_type_name) NOEXCEPT;
 #endif
 
-/**
- * @struct xen_domain_status
- * @brief Struct contain Xen-specific status flags
- * Field values are set based on the values in the
- * xen_domctl_getdomaininfo_t struct's flag field.
- */
-typedef struct xen_domain_status {
-    bool dying:1;                     /**< set if XEN_DOMINF_dying is set */
-    bool shutdown:1;                  /**< set if XEN_DOMINF_shutdown is set */
-    bool paused:1;                    /**< set if XEN_DOMINF_paused is set */
-    bool blocked:1;                   /**< set if XEN_DOMINF_blocked is set */
-    bool running:1;                   /**< set if XEN_DOMINF_running is set */
-    bool debugged:1;                  /**< set if XEN_DOMINF_debugged is set */
-    bool xs_domain:1;                 /**< set if XEN_DOMINF_xs_domain is set */
-    bool hardware_assisted_paging:1;  /**< set if XEN_DOMINF_hap is set */
-} xen_domain_status_t;
-
 /**
  * @struct domain_status
  * @brief Struct containing status flags for a domain.
- * Also contains hypervisor-specific flags in sub-structs
+ * Fields that are unsupported for the attached domain are set to -1.
+ * Fields that are true are set to 1.
+ * Fields that are false are set to 0.
  */
 typedef struct domain_status {
-    bool missing:1; /**< Indicates whether the domain was found */
-    vmi_mode_t type; /**< Hypervisor type, used to determine which fields exist in the union below */
-    union {
-        xen_domain_status_t xen_domain;
-    };
+    int dying:2;     /**< Set if the domain is dying */
```

Comment:
Please use `unsigned int` in the bitfield

Reply:
Could it be `signed int`, since I was thinking `-1` could be used in the event that a field was not supported on another hypervisor type?

Reply:
How would that work?

Reply:
I wasn't sure if all of these fields would make sense for all hypervisor types. Setting one of the fields to `-1` could be used to indicate that status flag is not applicable for a given hypervisor type. Maybe that's overthinking it though? I don't have a problem changing these to 1-bit unsigned int fields instead

Reply:
That sounds fine although I'm not sure under what circumstance would you actually check for that unsupported value for a status. Doing every time during run-time sounds like an unnecessary overhead. If its something you need to know it probably would be something you want to figure out at initialization time and set things up accordingly, no?

Reply:
Yeah, that's a good point. It also seems likely that any additional hypervisor drivers that implement support for this function would set status bits appropriately, even if (for instance) `dying` doesn't directly translate. I will update this to be unsigned int bitfields.